nano/node/node.cpp

```diff
@@ -1193,15 +1244,22 @@ class distributed_work : public std::enable_shared_from_this<distributed_work>
 		outstanding.erase (address);
 		return outstanding.empty ();
 	}
+	void add (std::shared_ptr<work_request> & connection)
+	{
+		std::lock_guard<std::mutex> lock (mutex);
+		connections.push_back (connection);
+	}
 	std::function<void(uint64_t)> callback;
 	unsigned int backoff; // in seconds
 	std::shared_ptr<nano::node> node;
 	nano::block_hash root;
 	std::mutex mutex;
 	std::map<boost::asio::ip::address, uint16_t> outstanding;
+	std::vector<std::weak_ptr<work_request>> connections;
 	std::vector<std::pair<std::string, uint16_t>> need_resolve;
 	std::atomic_flag completed;
 	uint64_t difficulty;
+	std::atomic_bool stopped{ false };
```

Comment:
We use `std::atomic<bool>` everywhere (mainly consistency now, the type used to be a problem on some compilers)